js/data/create_bucket.js

```diff
@@ -19,6 +19,21 @@ function createBucket(layer, buffers, z, overscaling, collisionDebug) {
         layout[k] = values[k].calculate(z, fakeZoomHistory);
     }
 
+    if (layer.type === 'symbol') {
+        // To reduce the number of labels that jump around when zooming we need
+        // to use a text-size value that is the same for all zoom levels.
+        // This calculates text-size at a high zoom level so that all tiles can
+        // use the same value when calculating anchor positions.
+        if (values['text-size']) {
+            layout['text-max-size'] = values['text-size'].calculate(18, fakeZoomHistory);
+            layout['text-size'] = values['text-size'].calculate(z + 1, fakeZoomHistory);
+        }
+        if (values['icon-size']) {
+            layout['icon-max-size'] = values['icon-size'].calculate(18, fakeZoomHistory);
+            layout['icon-size'] = values['icon-size'].calculate(z + 1, fakeZoomHistory);
+        }
```

Comment:
@ansis Are you sure this won't affect density much? In our template styles, we originally had used `-max-size` values that were always just equal to the largest value of a `-size` ramp (which is typically pretty close to being the z18 size), and found that this impacted the density of labels at low zoom levels (when the `-size` is relatively much smaller). That was the motivation for changing all the `-max-size` values to [use ramps that look something like this](https://github.com/mapbox/app/issues/492#issuecomment-74004095). From what you're saying  though it sounds like using a ramps for `-max-size` was not ideal?